SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUpstreamChecker.java

```diff
@@ -0,0 +1,197 @@
+    for (UpstreamInfo upstreamInSingularity : upstreamsInSingularityForService) {
+      final Collection<UpstreamInfo> matches = getEqualUpstreams(upstreamInSingularity, upstreamsInLoadBalancerForService);
+      upstreamsInLoadBalancerForService.removeAll(matches);
+    }
+    return new ArrayList<>(upstreamsInLoadBalancerForService);
+  }
+
+  private Collection<UpstreamInfo> getLoadBalancerUpstreamsForServiceHelper(SingularityCheckingUpstreamsUpdate singularityCheckingUpstreamsUpdate, Optional<String> loadBalancerUpstreamGroup) {
+    if (singularityCheckingUpstreamsUpdate.getBaragonServiceState().isPresent()){
+      LOG.trace("Baragon service state for service {} is present.", singularityCheckingUpstreamsUpdate.getSingularityRequestId());
+      final BaragonServiceState baragonServiceState = singularityCheckingUpstreamsUpdate.getBaragonServiceState().get();
+      return baragonServiceState.getUpstreams()
+          .stream()
+          .filter(upstream -> upstream.getGroup().equals(loadBalancerUpstreamGroup.or("default")))
+          .collect(Collectors.toList());
+    }
+    LOG.trace("Baragon service state for service {} is absent.", singularityCheckingUpstreamsUpdate.getSingularityRequestId());
+    return Collections.emptyList();
+  }
+
+  private Collection<UpstreamInfo> getLoadBalancerUpstreamsForService(String singularityRequestId, Optional<String> loadBalancerUpstreamGroup) {
+    try {
+      LOG.info("Sending request to get load balancer upstreams for service {}.", singularityRequestId);
+      final SingularityCheckingUpstreamsUpdate checkUpstreamsState = lbClient.getLoadBalancerServiceStateForRequest(singularityRequestId);
+      LOG.trace("Succeeded getting load balancer upstreams for singularity request {}. State is {}.", singularityRequestId, checkUpstreamsState.toString());
+      return getLoadBalancerUpstreamsForServiceHelper(checkUpstreamsState, loadBalancerUpstreamGroup);
+    } catch (Exception e) {
+      LOG.error("Failed getting load balancer upstreams for singularity request {}. ", singularityRequestId, e);
+    }
+    return Collections.emptyList();
+  }
+
+  private Optional<SingularityLoadBalancerUpdate> syncUpstreamsForServiceHelper(SingularityRequest singularityRequest, SingularityDeploy deploy, Optional<String> loadBalancerUpstreamGroup) {
+    try {
+      LOG.trace("Sending load balancer request to sync upstreams for service {}.", singularityRequest.getId());
+      final LoadBalancerRequestId loadBalancerRequestId = new LoadBalancerRequestId(String.format("%s-%s-%s", singularityRequest.getId(), deploy.getId(), System.currentTimeMillis()), LoadBalancerRequestType.REMOVE, Optional.absent());
+      Collection<UpstreamInfo> upstreamsInLoadBalancerForService = getLoadBalancerUpstreamsForService(singularityRequest.getId(), loadBalancerUpstreamGroup);
```

Comment:
are we taking into account the `SingularityDeploy.loadBalancerServiceIdOverride` here? That essentially lets something under one Singularity request id be under a different baragon service id.

e.g. MyService-web is normal and has group default, then MyService-userweb has `loadBalancerServiceIdOverride` set to MyService-web and `loadBalancerUpstreamGroup` set to `user` so that both share the same baragon service with different groups